test/task.cc

```diff
@@ -391,34 +416,40 @@ TEST(Task, Success) {
 }
 
 TEST(Task, Failure) {
-  auto e = []() -> Task::Of<std::string> {
+  auto e = []() -> Task::Of<std::string>::Raises<std::runtime_error> {
     return Task::Failure("error");
   };
 
+  static_assert(
+      eventuals::tuple_types_unordered_equals_v<
+          decltype(e())::ErrorsFrom<void, std::tuple<>>,
+          std::tuple<std::runtime_error>>);
+
   EXPECT_THROW_WHAT(*e(), "error");
 }
 
 TEST(Task, Inheritance) {
   struct Base {
-    virtual Task::Of<int> GetTask() = 0;
+    virtual Task::Of<int>::Raises<std::runtime_error> GetTask() = 0;
   };
 
   struct Sync : public Base {
-    Task::Of<int> GetTask() override {
-      return Task::Success(10);
+    Task::Of<int>::Raises<std::runtime_error> GetTask() override {
+      return Task::Success<int, std::runtime_error>(10);
```

Comment:
Let's make our API not require specifying the error type when we know we're successful!